modules/nlu/src/backend/application/training-queue.ts

```diff
@@ -251,7 +244,7 @@ export class TrainingQueue {
     await this._notify(trainId, this._fillSate({ status: 'errored' }))
 
     // eslint-disable-next-line @typescript-eslint/no-floating-promises
-    this._trainingService.transaction(repo => repo.set(trainId, this._fillSate({ status: 'needs-training' })))
+    this._trainingRepo.set(trainId, this._fillSate({ status: 'needs-training' }))
```

Comment:
mmmmhhhh 

I know we've talked about this, just to be sure : this is dangerous IMO. I would run in a transaction. What if another atomic get/set is occuring ?

Reply:
now in a transaction!